File: src/easy/count-students.ts
export const countStudents = (
    students: number[],
    sandwiches: number[],
): number => {
    let count = 0;
    while (students.length > 0 && sandwiches.length > 0) {
        if (count == sandwiches.length) break;
        if (students[0] == sandwiches[0]) {
            students.shift();
            sandwiches.shift();
            count = 0;
        } else {
            students.push(students.shift() ?? 0);
            count++;
        }
    }

    return students.length;
};

File: lib/easy/count_students.dart
int countStudents(List<int> students, List<int> sandwiches) {
  int rotations = 0;

  while (students.isNotEmpty && sandwiches.isNotEmpty) {
    if (rotations == students.length) break;
    if (students.first == sandwiches.first) {
      students.removeAt(0);
      sandwiches.removeAt(0);
      rotations = 0;
    } else {
      students.add(students.removeAt(0));
      rotations++;
    }
  }
  return students.length;
}
